lib/http.c

```diff
@@ -2847,16 +2848,17 @@ checkrtspprefix(struct Curl_easy *data,
   }
   if(CURLE_OK != Curl_convert_from_network(data, scratch, strlen(s)+1)) {
     /* Curl_convert_from_network calls failf if unsuccessful */
-    free(scratch);
-    return FALSE; /* can't return CURLE_foobar so return FALSE */
+    result = FALSE; /* can't return CURLE_foobar so return FALSE */
   }
-  s = scratch;
+  else
+    result = checkprefix("RTSP/", scratch)? TRUE: FALSE;
+  free(scratch);
 #else
   (void)data; /* unused */
+  result = checkprefix("RTSP/", s)? TRUE: FALSE;
 #endif /* CURL_DOES_CONVERSIONS */
-  if(checkprefix("RTSP/", s))
-    return TRUE;
-  return FALSE;
+
+  return result;
```

Comment:
This looks like a fix that is totally independent from the mime API work, right? If so I think we should cherry-pick it out from here and land it first, separately so that we don't make this work bigger than it has to be...

Reply:
Yes, I encountered a few of them during this work (like NULL data pointer in calling code conversion functions), and fixed them on the fly. Fixes are tiny however.